browser.c

```diff
@@ -1735,6 +1735,21 @@ void mutt_select_file(char *f, size_t flen, int flags, char ***files, int *numfi
               mutt_file_concat_path(tmp, LastDir, buf, sizeof(tmp));
               mutt_str_strfcpy(buf, tmp, sizeof(buf));
             }
+            /* resolve path. Avoids buildup such as /a/b/../../c
+             * Symlinks are always unraveled to keep code simple */
+            {
+              const char *real_path = NULL;
+              if ((real_path = realpath(buf, NULL)) == NULL)
```

Comment:
Also, I think we're more into style like this, right @flatcap ?
```c
const char *real_path = = realpath(buf, NULL);
if (real_path == NULL)
{
   ...
}
```

Reply:
Yes, please.
This is a simple case, where the code is legible, but this style can lead to some very hard-to-read conditionals.